dpdk/vr_dpdk_interface.c

```diff
@@ -979,11 +1012,30 @@ dpdk_if_rx(struct vr_interface *vif, struct vr_packet *pkt)
     rte_pktmbuf_dump(stdout, m, 0x60);
 #endif
 
+    vr_stats = vif_get_stats(tx_queue->q_vif, lcore_id);
     if (likely(tx_queue->txq_ops.f_tx != NULL)) {
         tx_queue->txq_ops.f_tx(tx_queue->q_queue_h, m);
+
+        /**
+         * Update counters for:
+         *  - packets enqueued to the interface successfully.
+         *  - packets which have been dropped when .f_tx() could not send.
+         */
+        if (likely(tx_queue->txq_ops.f_stats != NULL)) {
+            tx_queue->txq_ops.f_stats(tx_queue->q_queue_h, &port_stats, 0);
+
+            if (tx_queue->txq_ops.f_tx == rte_port_ring_writer_ops.f_tx) {
+                vr_stats->vis_iftxrngenqpkts = port_stats.n_pkts_in;
+                vr_stats->vis_iftxrngenqdrops = port_stats.n_pkts_drop;
+            } else {
+                vr_stats->vis_ifenqpkts = port_stats.n_pkts_in;
+                vr_stats->vis_ifenqdrops = port_stats.n_pkts_drop;
+            }
+        }
```

Comment:
MIght be good to put the above code in an inline function as it is called in multiple places.